### PR TITLE
UX Enhancements in Timings Module

### LINKED ISSUE #10: Better UX in Timings module
**Is your feature request related to a problem? Please describe.**
- Currently the app merely lists the timings which is a humongous task to search through.

**Describe the solution you'd like**
- A nice UX update would be to highlight the next bus/train time by checking current time.

### PR DESCRIPTION
**Pull request by ArunVigesh**
Fixes #10

### CODE REVIEW

File: app/src/main/java/in/co/rajkumaar/amritarepo/timings/ShuttleBusTimingsActivity.java [hunk 1]
```diff
@@ -161,21 +172,78 @@ private void loadData(String type) {
...
+                Calendar currentTime = Calendar.getInstance();
+                Log.e("TAG", "loadData: "+currentTime.getTime() );
+                ArrayList<String> timings = gson.fromJson(json, listType);
+                for (String item : timings) {
+                    Date busTimeDate=dateFormat.parse(item);
+                    Calendar busTime =Calendar.getInstance();
+                    busTime.setTime(new Date());
+                    busTime.set(Calendar.HOUR_OF_DAY,busTimeDate.getHours());
+                    busTime.set(Calendar.MINUTE,busTimeDate.getMinutes());
+                    Log.e("TAG", "loadData: "+busTime.getTime() );
```
Comment (rajkumaar23) [inline app/src/main/java/in/co/rajkumaar/amritarepo/timings/ShuttleBusTimingsActivity.java]: Can you remove all debug logs, except the error stack traces ?

File: app/src/main/java/in/co/rajkumaar/amritarepo/timings/PublicTransportsActivity.java [hunk 2]
```diff
@@ -245,119 +261,313 @@ public void onCancelled(@NonNull DatabaseError error) {
...
+                        nextTrainBus.setText("No Trains");
+                    }
+                } catch (JSONException | ParseException e) {
+                    e.printStackTrace();
                 }
             }
         }
 
         if (type != null && type.equals("Buses from Coimbatore")) {
+            trainbus.setImageResource(R.drawable.busimage);
```
Comment (rajkumaar23) [inline app/src/main/java/in/co/rajkumaar/amritarepo/timings/PublicTransportsActivity.java]: Why are there two files with same content, but different names?
Ain't `bus` and `busimage` the same files  ? Can you delete one ?

File: app/src/main/java/in/co/rajkumaar/amritarepo/timings/PublicTransportsActivity.java [hunk 3]
```diff
@@ -245,119 +261,313 @@ public void onCancelled(@NonNull DatabaseError error) {
...
-                            "cbe",
-                            "etmd",
-                            "bus"
-                    ));
+                try {
+                    Calendar currentTime = Calendar.getInstance();
+                    Log.e("TAG", "loadData: " + currentTime.getTime());
+                    for (int i = 0; i < timings.length(); ++i) {
+                        JSONObject item = timings.getJSONObject(i);
+                        Date busTimeDate = dateFormat.parse(item.getString("dep"));
```
Comment (rajkumaar23) [inline app/src/main/java/in/co/rajkumaar/amritarepo/timings/PublicTransportsActivity.java]: Can this piece of code be written in a new function and called from here?

File: app/src/main/java/in/co/rajkumaar/amritarepo/timings/PublicTransportsActivity.java [hunk 4]
```diff
@@ -245,119 +261,313 @@ public void onCancelled(@NonNull DatabaseError error) {
...
+                                item.getString("dep"),
+                                "etmd",
+                                "pkd",
+                                "train"
+                        ));
+                        if(currentTime.get(Calendar.DAY_OF_WEEK)==1 && item.getString("days").equals("All Days Except Sundays"))
+                        {
+                            continue;
+                        }
+                        else if(currentTime.get(Calendar.DAY_OF_WEEK)==5 && item.getString("days").equals("All Days Except Thursdays"))
```
Comment (rajkumaar23) [inline app/src/main/java/in/co/rajkumaar/amritarepo/timings/PublicTransportsActivity.java]: Can you write a more generic logic for this ? Like get current `DAY` in string as for eg : `Friday` and check if `item.getString("days").toLowerCase().contains(currentDay.toLowerCase())`?

File: app/src/main/java/in/co/rajkumaar/amritarepo/timings/PublicTransportsActivity.java [hunk 5]
```diff
@@ -245,119 +261,313 @@ public void onCancelled(@NonNull DatabaseError error) {
...
+                            flag=1;
+                            Date startTime=currentTime.getTime();
+                            Date endTime=trainTime.getTime();
+                            long timediff=endTime.getTime()-startTime.getTime();
+                            countdown(timediff);
+                        }
+                    }
+                    if(flag==0)
+                    {
+                        nextTrainBus.setText("No Trains");
```
Comment (rajkumaar23) [inline app/src/main/java/in/co/rajkumaar/amritarepo/timings/PublicTransportsActivity.java]: Please modularize the code. :wink:

File: app/src/main/java/in/co/rajkumaar/amritarepo/timings/PublicTransportsActivity.java [hunk 6]
```diff
@@ -433,5 +643,55 @@ public void onClick(DialogInterface dialogInterface, int i) {
         }
         return super.onOptionsItemSelected(item);
     }
+    private void countdown(long timeDiff)
```
Comment (rajkumaar23) [inline app/src/main/java/in/co/rajkumaar/amritarepo/timings/PublicTransportsActivity.java]: Whenever you commit files, ensure you `Reformat Code` to maintain code standards and good indendation.

File: app/src/main/java/in/co/rajkumaar/amritarepo/timings/PublicTransportsActivity.java [hunk 7]
```diff
@@ -52,30 +58,40 @@
...
-
-
+        countdownTimer=findViewById(R.id.countdownTime);
+        nextTrainBus=findViewById(R.id.nextBusTrainTime);
         listView = findViewById(R.id.timings_list);
         Bundle extras = getIntent().getExtras();
-        final String type = extras.getString("type");
-
+        type = extras.getString("type");
+        trainbus=findViewById(R.id.busTrainLogo);
```
[the remark this answers is not in the document]
Comment (ArunVigesh) [reply app/src/main/java/in/co/rajkumaar/amritarepo/timings/PublicTransportsActivity.java]: Ok.. Will do the Needful

File: app/src/main/java/in/co/rajkumaar/amritarepo/timings/PublicTransportsActivity.java (hunk 2 above)
Comment (ArunVigesh) [reply app/src/main/java/in/co/rajkumaar/amritarepo/timings/PublicTransportsActivity.java]: Used Separate Images for both Class

Comment (rajkumaar23) [reply app/src/main/java/in/co/rajkumaar/amritarepo/timings/PublicTransportsActivity.java]: Oh! Alright then. :+1:

**Comment by rajkumaar23** [comment]
@all-contributors please add @ArunVigesh  for design, code

**Pull request merged**